pints/_plot.py

```diff
@@ -0,0 +1,248 @@
+#
+# Quick diagnostic tools for inference results
+#
+# This file is part of PINTS.
+#  Copyright (c) 2017, University of Oxford.
+#  For licensing information, see the LICENSE file distributed with the PINTS
+#  software package.
+#
+import pints
+import numpy as np
+from scipy import stats
+#import matplotlib.pyplot as plt
+import pylab as plt
+
+def trace(chain):
+    """
+    Trace plot.
+
+    `chain`
+        MCMC routine returned chain
+
+    """
+    n_sample, n_param = chain.shape
+
+    fig = plt.figure(figsize=(12, 2*n_param))
+    for i in xrange(n_param):
+        # Add histogram subplot
+        plt.subplot(n_param, 2, 1 + 2 * i)
+        plt.xlabel('Parameter ' + str(i + 1))
+        plt.ylabel('Frequency')
+        plt.hist(chain[:,i], label='p' + str(i + 1), bins=40, color='green',
+        alpha=0.5)
+
+        # Add trace subplot
+        plt.subplot(n_param, 2, 2 + 2 * i)
+        plt.xlabel('Iteration')
+        plt.ylabel('Parameter ' + str(i + 1))
+        plt.plot(chain[:,i], color='green', alpha=0.5)
+
+    plt.tight_layout()
+    return fig
+
+def autocorrelation(chain):
+    """
+    Autocorrelation plot for MCMC.
+
+    `chain`
+        MCMC routine returned chain
+
+    """
+    n_sample, n_param = chain.shape
+
+    fig, axes = plt.subplots(n_param, 1, sharex=True, figsize=(6, 2*n_param))
+    for i in xrange(n_param):
+        axes[i].acorr(chain[:,i] - np.mean(chain[:,i]), label='p'+str(i+1))
+        axes[i].set_xlim(-0.5, 20.5)
+    axes[i].set_xlabel('Lag')
+    fig.text(0.04, 0.5, 'Autocorrelation', va='center', rotation='vertical')
+
+    return fig
+
+def prediction(chain, problem):
+    """
+    Predicted time series plot.
+
+    `chain`
+        MCMC routine returned chain
+    `problem`
+        pints.SingleSeriesProblem object
+
+    """
+    if not isinstance(problem, pints.SingleSeriesProblem):
+        raise TypeError('pints.SingleSeriesProblem type is expected')
+
+    n_sample, n_param = chain.shape
+    plot_n_sample = 1000 if n_sample>1000 else n_sample
+    times = problem.times()
+
+    # Evaluate the model for all inferred parameters
+    predicted_values = []
+    for params in chain[-plot_n_sample:,:]:
+        predicted_values.append(problem.evaluate(params[:-1]))
+    predicted_values = np.array(predicted_values)
+    mean_values = np.mean(predicted_values, axis=0)
+
+    # Plot prediction
+    fig = plt.figure(figsize=(7.5, 3.75))
+    plt.xlabel('Time')
+    plt.ylabel('Value')
+    plt.plot(times, predicted_values[0], color='#1f77b4', label='inferred series')
+    for v in predicted_values[1:]:
+        plt.plot(times, v, color='#1f77b4', alpha=0.05)
+    plt.plot(times, mean_values, color='black', lw=2, label='mean inferred')
+    plt.plot(times, problem.values(), 'o', color='#7f7f7f', ms=6.5, label='data points')
+
+    return fig
+
+def forceAspect(ax, aspect=1):
```

Comment:
Please stick to `lower_case_names_with_underscores` for methods and functions! (ClassesAreAllCamelCased)

Reply:
Actually, I would make this a hidden method `_force_aspect` and maybe give it a better name `_force_equal_aspect_ratio` for example